### DISCUSSION TITLE
CSRF error in IFrame

### DISCUSSION THREAD

**Discussion by saboteurkid** [ory/hydra#2670] [Q&A] [answered]
I got "No CSRF value available in the session cookie." while using Hydra in Iframe even if I had `SERVE_COOKIES_SAME_SITE_MODE=None` and `SERVE_COOKIES_SAME_SITE_LEGACY_WORKAROUND=true`

Context:

```
https://myapp.com
    |_ iframe: https://3rd.app
       Redirect to: https://my.hydra/auth
       Redirect to: https://my.hydra/login
       Redirect to: https://my.hydra/auth
       Redirect to: https://3rd.app/callback?error=request_forbidden&error_description=The request is not allowed. No CSRF value available in the session cookie.
```

What would I do to get over this issue?

Thanks

**Comment by saboteurkid** [comment, marked as answer]
I've debugged, the `oauth2_authentication_csrf_insecure` cookie does not include in 2nd `https://my.hydra/auth`. 

`oauth2_authentication_csrf_insecure` cookie was not stored by the browser.

I don't quite understand the behavior of browser (FF).

**Reply by vinckr** [reply]
Hey @saboteurkid , just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
